src/eve/utils.py

```diff
@@ -281,3 +281,361 @@ def reset_sequence(cls, start: int = 1) -> None:
+
+        - ``@``: pick data from each item in a sequence (equivalent to ``toolz.itertoolz.pluck(index, iterator)``).
+          For detailed information check :func:`toolz.itertoolz.pluck` reference.
+
+            >>> it = xiter([('a', 1), ('b', 2), ('c', 3)])
+            >>> list(it @ 0)
+            ['a', 'b', 'c']
+
+            >>> it = xiter([
+            ...     dict(name="AA", age=20, country="US"),
+            ...     dict(name="BB", age=30, country="UK"),
+            ...     dict(name="CC", age=40, country="CH")
+            ... ])
+            >>> list(it @ ["name", "age"])
+            [('AA', 20), ('BB', 30), ('CC', 40)]
+
+        - ``+``: chain iterators (equivalent to ``itertools.chain(it_a, it_b)``).
+          For detailed information check :func:`itertools.chain` reference.
+
+            >>> it_a, it_b = xiter(range(2)), xiter(['a', 'b'])
+            >>> list(it_a + it_b)
+            [0, 1, 'a', 'b']
+
+            >>> it_b = xiter(['a', 'b'])
+            >>> list([0, 1] + it_b)
+            [0, 1, 'a', 'b']
+
+        - ``-``: diff iterators (equivalent to ``toolz.itertoolz.diff(it_a, it_b)``).
+          For detailed information check :func:`toolz.itertoolz.diff` reference.
+
+            >>> it_a, it_b = xiter([1, 2, 3]), xiter([1, 3, 5])
+            >>> list(it_a - it_b)
+            [(2, 3), (3, 5)]
+
+            >>> it_b = xiter([1, 3, 5])
+            >>> list([1, 2, 3] - it_b)
+            [(2, 3), (3, 5)]
+
+            Adding missing values:
+
+            >>> it_a, it_b = xiter([1, 2, 3, 4]), xiter([1, 3, 5])
+            >>> list(it_a - xiter_args(it_b, default=None))
+            [(2, 3), (3, 5), (4, None)]
+
+            >>> it_b = xiter([1, 3, 5])
+            >>> list(xiter_args([1, 2, 3, 4], default=None) - it_b)
+            [(2, 3), (3, 5), (4, None)]
+
+
+        - ``*``: product of iterators (equivalent to ``itertools.product(it_a, it_b)``).
+          For detailed information check :func:`itertools.product` reference.
+
+            >>> it_a, it_b = xiter([0, 1]), xiter(['a', 'b'])
+            >>> list(it_a * it_b)
+            [(0, 'a'), (0, 'b'), (1, 'a'), (1, 'b')]
+
+            >>> it_b = xiter(['a', 'b'])
+            >>> list([0, 1] * it_b)
+            [(0, 'a'), (0, 'b'), (1, 'a'), (1, 'b')]
+
+            Product of an iterator with itself:
+
+            >>> it_a = xiter([0, 1])
+            >>> list(it_a * 3)
+            [(0, 0, 0), (1, 1, 1)]
+
+            >>> it_a = xiter([0, 1])
+            >>> list(3 * it_a)
+            [(0, 0, 0), (1, 1, 1)]
+
+        - ``/``: partition iterator into tuples of length at most ``n``
+          (equivalent to ``toolz.itertoolz.partition_all(n, iterator)``).
+          For detailed information check :func:`toolz.itertoolz.partition_all` reference.
+
+            >>> it = xiter(range(7))
+            >>> list(it / 3)
+            [(0, 1, 2), (3, 4, 5), (6,)]
+
+        - ``//``: partition iterator into tuples of length ``n``
```

Comment:
I am wondering if it is a good idea to introduce non standard operators here. As far as I understand tooling will not help here. What I mean: if you see the following code `a // 3` somewhere. How do you figure out what it does (without knowing all of eve).

Reply:
Yes, I agree, it could too confusing to undertand the code using operators with arbitrary meanings. I've changed the PR to use proper method names, which can also be chained and they are easier to read.